mobile/mobile-core/deviceEmitter.ts

```diff
@@ -0,0 +1,35 @@
+///<reference path="./../../.d.ts"/>
+"use strict";
+
+import { EventEmitter } from "events";
+
+import fiberBootstrap = require("../../fiber-bootstrap");
+
+class DeviceEmitter extends EventEmitter {
+	// TODO: add iOSDeviceDiscovery as a dependency too
+	constructor(private $androidDeviceDiscovery:Mobile.IAndroidDeviceDiscovery,
+		private $devicesService: Mobile.IDevicesService) {
+		super();
+	}
+
+	public initialize(): IFuture<void> {
+		return (() => {
+			this.$androidDeviceDiscovery.ensureAdbServerStarted().wait();
+			this.$androidDeviceDiscovery.on("deviceFound", (data: Mobile.IDevice) => {
+				this.emit("deviceFound", data.deviceInfo);
+			});
+
+			this.$androidDeviceDiscovery.on("deviceLost", (data: Mobile.IDevice) => {
+				this.emit("deviceLost", data.deviceInfo);
+			});
+
+			setInterval(() =>
+				fiberBootstrap.run(() => {
+					this.$androidDeviceDiscovery.startLookingForDevices().wait();
+				}),
+			500).unref();
```

Comment:
This 500 should be a named constant. I am guilty of introducing this magic number :(